server/service.go

```diff
@@ -299,29 +300,37 @@ func (m *BoostService) handleGetHeader(w http.ResponseWriter, req *http.Request)
 		ua            = UserAgent(req.Header.Get("User-Agent"))
 	)
 
+	// Parse the slot
 	slotValue, err := strconv.ParseUint(vars["slot"], 10, 64)
 	if err != nil {
 		m.respondError(w, http.StatusBadRequest, errInvalidSlot.Error())
 		return
 	}
 	slot := phase0.Slot(slotValue)
 
+	// Add relevant fields to the logger
 	log := m.log.WithFields(logrus.Fields{
 		"method":     "getHeader",
 		"slot":       slot,
 		"parentHash": parentHashHex,
 		"pubkey":     pubkey,
 		"ua":         ua,
 	})
-	log.Debug("getHeader")
+	log.Debug("handling request")
+
+	// Additional header fields
+	header := req.Header
+	header.Set("User-Agent", wrapUserAgent(ua))
+	header.Set(HeaderStartTimeUnixMS, fmt.Sprintf("%d", time.Now().UTC().UnixMilli()))
```

Comment:
Not a huge fan of doing a bunch of request specific stuff here, feels it should be done inside `m.getHeader` for not leaking and consitency.